src/ProjectOrigin.WalletSystem.Server/CommandHandlers/VerifySliceCommandHandler.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using MassTransit;
+using Microsoft.Extensions.Logging;
+using ProjectOrigin.WalletSystem.Server.Activities.Exceptions;
+using ProjectOrigin.WalletSystem.Server.Database;
+using ProjectOrigin.WalletSystem.Server.Extensions;
+using ProjectOrigin.WalletSystem.Server.Models;
+using ProjectOrigin.WalletSystem.Server.Projections;
+using ProjectOrigin.WalletSystem.Server.Services;
+
+namespace ProjectOrigin.WalletSystem.Server.CommandHandlers;
+
+public record VerifySliceCommand(Guid Id,
+                    Guid DepositEndpointId,
+                    int DepositEndpointPosition,
+                    string Registry,
+                    Guid CertificateId,
+                    long Quantity,
+                    byte[] RandomR);
+
+public class VerifySliceCommandHandler : IConsumer<VerifySliceCommand>
+{
+    private readonly ILogger<VerifySliceCommandHandler> _logger;
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IRegistryService _registryService;
+
+    public VerifySliceCommandHandler(ILogger<VerifySliceCommandHandler> logger, IUnitOfWork unitOfWork, IRegistryService registryService)
+    {
+        _logger = logger;
+        _unitOfWork = unitOfWork;
+        _registryService = registryService;
+    }
+
+    public async Task Consume(ConsumeContext<VerifySliceCommand> context)
+    {
+        var receivedSlice = context.Message;
+
+        // Get Granular Certificate Projection from registry
+        var getCertificateResult = await _registryService.GetGranularCertificate(receivedSlice.Registry, receivedSlice.CertificateId);
```

Comment:
In case ` _registryService.GetGranularCertificate` throws an exception, will MassTransit retry this message?

Reply:
No, it will be moved to the _error queue, since all though scenarios will return a object, since the exception would most likely continue until a code change.